fecgov/fec-cms
# 6570 - Icons and meta data

## Summary

- Resolves #6570 
- Resolves #6721 

(Include a summary of proposed changes and connect issue below)

### Required reviewers

- 1 front-end?
- 1 design/UX?

## Impacted areas of the application

Meta data site-wide and various social sharing images and thumbnails:
- New devices and resolutions = new image sizes and dropping older image sizes
- New iOS: renaming 'apple-touch-icon-precomposed' ([now many years abandoned](https://webhint.io/docs/user-guide/hints/hint-apple-touch-icons/)) to the more standard `apple-touch-icon`
- Dropped [outdated Internet Explorer tags](https://learn.microsoft.com/en-us/windows-hardware/customize/desktop/unattend/microsoft-windows-helpandsupport-helpandsupport-tilecolor)
- Dropped outdated [`HandheldFriendly`](https://getoutofmyhead.dev/handheld-friendly/) and [`MobileOptimized`](https://www.google.com/search?q=do+we+need+mobileoptimized+meta+tag) meta data tags (used for very old versions of BlackBerry and Windows Mobile devices)
- For the `/general/favicons-*.png`, updating to "FEC.gov" instead of "The new FEC.gov" since the site's been standard for more than five years
- Removed the "This site is in beta" message from the top of the header/seal image that only appears when printing pages
- Added a white background to the smallest favicon since dark mode is a thing across devices
- Removed `START_YEAR` and `END_YEAR` window vars since the front-end isn't referencing them inside JavaScript

## Screenshots

The new images are in Files Changed but otherwise won't show up until this is merged, deployed, and crawled again

## Related PRs

There's another issue/PR for the canonical tag so we aren't touching those here.

## How to test

- Pull the branch
- `npm run build`
- Find someplace JavaScript is using `START_YEAR`?
- Find someplace JavaScript is using `END_YEAR`?
- Find someplace JavaScript is using `TRANSITION_URL`?
- Review the image changes shown here i.e. the ones added and removed listed under Files Changed
- For
   - Homepage
   - About
   - H4CC page(s)
   - Some data pages
   - Some legal pages
- Check
   - That the source code's `<link rel="icon"` `href` values are accurate
   - That the source code's `<link rel="apple-touch-icon"` `href` values are accurate
